### DISCUSSION TITLE
Can MibS solve Parametric Discrete Bilevel Problems ?

### DISCUSSION THREAD

**Discussion by ansarimoiz** [coin-or/MibS#98] [Q&A]
Hello,

I have a bilevel problem in which the lower level variable is discrete.
The objective function and constraint of the lower level involves product of upper level and lower level variables.
The upper level variable is continuous.

I was wondering whether such a problem can be solved by using MibS ?
Or is there any such solver which would be able to do it. 

Thank You

**Comment by tkralphs** [comment]
Sorry for the delay in answering. MibS would not be able to solve an instance of this type, both because you have continuous linking variables and because there are products of variables in the lower-level, which means the relaxation would be a non-linear optimization problem. In general, problems with continuous linking variables may not even have an optimum (i.e., the feasible set may be open and the infimum/supremum may not be attained by any solution), so you would need to know something about your model to assure that it even has a solution. Of course, you can always optimize over the closure of the feasible region as an approximation, but this can be arbitrarily far off. There are a number of algorithms for solving general non-linear bilevel problems, but I haven't kept up on which ones have open source implementations available.

**Reply by shabnamvaziri** [reply]
Dear @tkralphs ,
I am wondering if Mibs can solve the case when the objective function of the lower level involves the product of binary upper-level and continuous lower-level variables. The upper-level and lower-level models are mixed-integer. I have problem running the model as I face the following error: 
![Mibs](https://github.com/coin-or/MibS/assets/112642990/85787304-d61e-42ca-ab60-c3d3dddaaa10)

As I have MIP in both levels, I thought the problem might be related to this post. 
Thank you in advance.

**Comment by ansarimoiz** [comment]
Thank you for a detailed reply @tkralphs 
Do update me if you come across any algorithm for solving non-linear bilevel problems.
